Add distance filtering to region monitoring

rclee opened the pull request:
Testing required. 

ref: https://github.com/mapbox/mapbox-events-ios/issues/155

rclee committed:
Add distance filtering to region monitoring

alfwatt commented:
Please add:
- Unit Test Cases 
- Test Plan (probably needs a GPX file for the simulator with a bunch of points inside the area and another with points that eventually lead outside of the area)
- ~~Configurable radii for both the fence and the filter~~

Commit:
Implement NSUserDefaults+MMEConfiguration and Tests

https://github.com/mapbox/mobile-telemetry/issues/448

Commit:
Separate Cedar tests from XCTest test

https://github.com/mapbox/mobile-telemetry/issues/450

Commit:
🔨 Integrate MMEConfiguration into MapboxMobileEvents #456

https://github.com/mapbox/mobile-telemetry/issues/456

- Remove `MMEEventsConfiguration` and `MMEPinningConfigurationProvider`
- Adds MMERunningLock for all your background task completion timer needs
- Adds `mme_registerDefaults` to NSUserDefaults+MMEConfiguration for greater testability

rclee committed:
Merge branch 'master' into rclee-issue-155

alfwatt commented:
Stale
